lib/string-checker.js

```diff
@@ -126,6 +133,41 @@ StringChecker.prototype = {
     },
 
     /**
+     * Fix provided error.
+     *
+     * @param {JsFile} file
+     * @param {Errors} errors
+     * @protected
+     */
+    _fixJsFile: function(file, errors) {
+        var list = errors.getErrorList();
+        var configration = this.getConfiguration();
+
+        list.forEach(function(error) {
+            if (error.fixed) {
+                return;
+            }
+
+            var instance = configration.getConfiguredRule(error.rule);
+
+            if (instance && instance.fix) {
+                try {
+
+                    // "error.fixed = true" should go first, so rule can
+                    // decide for itself (with "error.fixed = false")
+                    // if it can fix this particular error
+                    error.fixed = true;
```

Comment:
For situation when rule decides it can't fix the error, if case is too complicated, for exampe. Then in the `fix` method it can do `error.fixed = false`.